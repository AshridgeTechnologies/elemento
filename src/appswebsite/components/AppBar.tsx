import * as React from 'react';
import Mui_AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import IconButton from '@mui/material/IconButton';
import MenuIcon from '@mui/icons-material/Menu';
import UserMenu from '../../shared/UserMenu'

export default function AppBar() {
    return (
        <Box sx={{ flexGrow: 1 }}>
            <Mui_AppBar position="sticky">
                <Toolbar variant="dense">
                    <IconButton
                        size="large"
                        edge="start"
                        color="inherit"
                        aria-label="menu"
                        sx={{ mr: 2 }}
                    >
                        <MenuIcon />
                    </IconButton>
                    <Typography variant="h6" component="div" sx={{ fontSize: 16, flexGrow: 1 }}>
                        Elemento Apps
                    </Typography>
                    <UserMenu/>
                </Toolbar>
            </Mui_AppBar>
        </Box>
    );
}