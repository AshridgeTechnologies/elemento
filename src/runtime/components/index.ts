export {default as App} from './App'
export {default as Button} from './Button'
export {default as Icon} from './Icon'
export {default as Image} from './Image'
export {default as UserLogon} from './UserLogon'
export {default as Menu} from './Menu'
export {default as MenuItem} from './MenuItem'
export {default as NumberInput} from './NumberInput'
export {default as DateInput} from './DateInput'
export {default as TrueFalseInput} from './TrueFalseInput'
export {default as SpeechInput} from './SpeechInput'
export {default as Data} from './Data'
export {default as Calculation} from './Calculation'
export {default as Timer} from './Timer'
export {default as Collection} from './Collection'
export {default as Page} from './Page'
export {default as ComponentElement} from './ComponentElement'
export {default as Form} from './Form'
export {default as BaseFormState} from './FormState'
export {default as SelectInput} from './SelectInput'
export {default as TextElement} from './TextElement'
export {default as Layout} from './Layout'
export {default as Block} from './Block'
export {default as AppBar} from './AppBar'
export {default as ListElement} from './ListElement'
export {default as ListItem} from './ListItem'
export {default as ItemSet} from './ItemSet'
export {default as ItemSetItem} from './ItemSetItem'
export {default as TextInput} from './TextInput'
export {default as MemoryDataStore} from './MemoryDataStore'
export {default as FileDataStore} from './FileDataStore'
export {default as FileDataStoreImpl} from './FileDataStoreImpl'
export {default as IdbDataStoreImpl} from './IdbDataStoreImpl'
export {default as BrowserDataStore} from './BrowserDataStore'
export {default as FirestoreDataStore} from './FirestoreDataStore'
export {default as FirestoreDataStoreImpl} from './FirestoreDataStoreImpl'
export {default as ServerAppConnector} from './ServerAppConnector'
export {default as Adapter} from './Adapter'
