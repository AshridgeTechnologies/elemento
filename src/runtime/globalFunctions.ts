import {fromPairs, isNil, last, sort, sortBy, splitEvery} from 'ramda'
import {
    add,
    differenceInCalendarDays,
    differenceInDays,
    differenceInHours,
    differenceInMinutes,
    differenceInMonths,
    differenceInSeconds,
    differenceInYears,
    format,
    parseISO
} from 'date-fns'
import * as csv from 'csv-parse/sync'
import {Value, valueOf, valuesOf} from './runtimeFunctions'
import {isNumeric, noSpaces} from '../util/helpers'
import {CastingContext} from 'csv-parse/lib'

type TimeUnit = 'seconds' | 'minutes' | 'hours' | 'days' | 'months' | 'years'
const unitTypes = ['seconds' , 'minutes' , 'hours' , 'days' , 'months' , 'years']

export const globalFunctions = {
    valueOf,

    Sum(...args: Value<number>[]) {
        return args.reduce( (acc, val) => (acc as number) + (val as number), 0)
    },

    Log(...args: any[]) {
        console.log(...args)
    },

    If(condition: any, trueValue: any, falseValue?: any) {
        return valueOf(condition) ? trueValue : falseValue
    },

    Left(s: Value<string>, length: Value<number>) {
        return valueOf(s).substring(0, valueOf(length))
    },

    Mid(s: Value<string>, start: Value<number>, length?: Value<number>) {
        const sVal = valueOf(s), startVal = valueOf(start), lengthVal = valueOf(length)
        if (startVal < 1) throw new Error(`Function Mid parameter 2 (start) is ${startVal}. It should be greater than or equal to 1.`)
        if (lengthVal !== undefined && lengthVal < 0) throw new Error(`Function Mid parameter 3 (length) is ${lengthVal}. It should be greater than or equal to 0.`)
        return sVal.substring(startVal - 1, startVal - 1 + (lengthVal ?? sVal.length))
    },

    Right(s: Value<string>, length: Value<number>) {
        const sVal = valueOf(s), lengthVal = valueOf(length)
        return sVal.substring(sVal.length - lengthVal)
    },

    And(...args: Value<any>[]) {
        return args.reduce( (prev, curr) => prev && !!valueOf(curr), true )
    },

    Or(...args: Value<any>[]) {
        return args.reduce( (prev, curr) => prev || !!valueOf(curr), false )
    },

    Not(arg: Value<any>) {
        return !valueOf(arg)
    },

    Substitute(s: Value<string>, toReplace: Value<string>, replaceWith: Value<string>) {
        const sVal = valueOf(s), toReplaceVal = valueOf(toReplace), replaceWithVal = valueOf(replaceWith)
        if (toReplaceVal === '') return sVal
        return sVal.replace(new RegExp(toReplaceVal, 'g'), replaceWithVal)
    },

    Max(...args: Value<number>[]) {
        if (args.length === 0) throw new Error('Wrong number of arguments to Max. Expected at least 1 argument.')
        return Math.max(...args as number[])
    },

    Min(...args: Value<number>[]) {
        if (args.length === 0) throw new Error('Wrong number of arguments to Min. Expected at least 1 argument.')
        return Math.min(...args as number[])
    },

    Record(...args: Value<any>[]) {
        if (args.length % 2 !== 0) throw new Error('Odd number of arguments - must have pairs of name, value')
        const argVals = args.map( valueOf )
        const pairs = splitEvery(2, argVals) as [string, any][]
        return fromPairs(pairs)
    },

    List(...args: Value<any>[]) {
        return args.map( valueOf )
    },

    Select(list: Value<any[]>, condition: (item: any) => boolean) {
        const listVal = valueOf(list) ?? []
        if (condition === undefined) throw new Error('Wrong number of arguments to Select. Expected list, expression.')
        return listVal.filter(condition)
    },

    ForEach(list: Value<any[]>, transform: (item: any) => any) {
        const listVal = valueOf(list) ?? []
        if (transform === undefined) throw new Error('Wrong number of arguments to ForEach. Expected list, expression.')
        return listVal.map(transform)
    },

    First(list: Value<any[]>, condition: (item: any) => boolean = () => true) {
        const listVal = valueOf(list) ?? []
        if (list === undefined) throw new Error('Wrong number of arguments to First. Expected list, optional expression.')
        return listVal.filter(condition)[0] ?? null
    },

    Last(list: Value<any[]>, condition: (item: any) => boolean = () => true) {
        const listVal = valueOf(list) ?? []
        if (list === undefined) throw new Error('Wrong number of arguments to Last. Expected list, optional expression.')
        return last(listVal.filter(condition)) ?? null
    },

    Sort(list: Value<any[]>, sortKeyFn: (item: any) => any | any[]): any[] {
        const compareItems = (a: any, b: any): -1 | 0 | 1 => {
            const aa = sortKeyFn(a), bb = sortKeyFn(b)
            return compareValues(aa, bb)
        }

        const compareValues = (a: any, b: any): -1 | 0 | 1 => {
            if (isNil(a) && isNil(b)) {
                return 0
            } else if (isNil(a)) {
                return -1
            } else if (isNil(b)) {
                return 1
            }
            if (Array.isArray(a) && Array.isArray(b)) {
                return compareArrays(a, b)
            }
            return a < b ? -1 : a > b ? 1 : 0
        }

        const compareArrays = (a: any[], b: any[]) => {
            for (let i = 0; i < a.length; i++) {
                const compareVal = compareValues(a[i], b[i])
                if (compareVal !== 0) {
                    return compareVal
                }
            }
            return 0
        }

        const listVal = valueOf(list)
        if (isNil(listVal)) {
            return listVal
        }
        return sort(compareItems, listVal)
    },

    Timestamp() {
        return Date.now()
    },

    Now() {
        return new Date()
    },

    Today(date = new Date()) {
        return new Date(Date.UTC(date.getFullYear(), date.getMonth(), date.getDate() ))
    },

    TimeBetween(date1Val: Value<Date>, date2Val: Value<Date>, unitVal: Value<TimeUnit>) {
        const [date1, date2, unit] = valuesOf(date1Val, date2Val, unitVal)
        if (!date1 || !date2) return undefined
        const unitTypes = ['seconds' , 'minutes' , 'hours' , 'days' , 'months' , 'years']
        switch(unit) {
            case 'seconds': return differenceInSeconds(date2, date1)
            case 'minutes': return differenceInMinutes(date2, date1)
            case 'hours': return differenceInHours(date2, date1)
            case 'days': return differenceInDays(date2, date1)
            case 'months': return differenceInMonths(date2, date1)
            case 'years': return differenceInYears(date2, date1)
            default: throw new Error(`Unknown unit ${unit}.  Should be one of ${unitTypes.join(', ')}`)
        }

    },

    DaysBetween(date1Val: Value<Date>, date2Val: Value<Date>) {
        const [date1, date2] = valuesOf(date1Val, date2Val)
        if (!date1 || !date2) return undefined
        return differenceInCalendarDays(date2, date1)
    },

    DateAdd(dateVal: Value<Date>, changeVal: Value<number>, unitVal: Value<TimeUnit>) {
        const [date, change, unit] = valuesOf(dateVal, changeVal, unitVal)
        if (!unitTypes.includes(unit)) {
            throw new Error(`Unknown unit ${unit}.  Should be one of ${unitTypes.join(', ')}`)
        }
        if (!date || change === undefined || change === null) return undefined

        return add(date, {[unit]: change})
    },

    DateFormat(date: Value<Date>, pattern: string) {
        if (!date) return ''
        return format(valueOf(date), pattern)
    },

    CsvToRecords(csvText: string, columnNames?: string[]) {
        const cast = (field: string, context: CastingContext) => {
            if (context.quoting) return field
            if (field === '') return undefined
            if (isNumeric(field)) return Number(field)
            if (field.toLowerCase() === 'true') return true
            if (field.toLowerCase() === 'false') return false
            const date = parseISO(field)
            if (!Number.isNaN(date.getTime())) {
                return date
            }
            return field
        }
        const convertColumnNames = (names: string[]) => names.map( noSpaces )
        const removeUndefinedFields = (obj: object) => {
            const definedEntries = Object.entries(obj).filter( ([, value]) => value !== undefined)
            return Object.fromEntries(definedEntries)
        }
        const lines = csvText.split(/[\r\n]+/)
        const tabDelimiters = lines.every( line => line.includes('\t'))
        return csv.parse(csvText, {
            columns: columnNames ? convertColumnNames(columnNames): convertColumnNames,
            delimiter: tabDelimiters ? '\t' : ',',
            cast,
            skip_empty_lines: true,
            skip_records_with_empty_values: true,
            relax_quotes: true,
            relax_column_count: true,
            trim: true
        }).map(removeUndefinedFields)
    }
}

export const functionArgIndexes = {
    Select: [1],
    ForEach: [1],
    First: [1],
    Last: [1],
    Sort: [1],
}