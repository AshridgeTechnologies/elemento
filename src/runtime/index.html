<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="initial-scale=1, width=device-width" />
  <title>Elemento App</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"/>
</head>
<body>
  <div id="main"></div>
  <script src="https://unpkg.com/react@17/umd/react.development.js" crossorigin></script>
  <script src="https://unpkg.com/react-dom@17/umd/react-dom.development.js" crossorigin></script>
  <script src="runApp.ts" type="module"></script>
</body>
</html>