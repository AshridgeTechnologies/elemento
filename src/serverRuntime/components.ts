export {default as FirestoreDataStore} from './FirestoreDataStore'
export {default as Collection} from './Collection'

// NOTE: Must add the name of each component to ./names.ts