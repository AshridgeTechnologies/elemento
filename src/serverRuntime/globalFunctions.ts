export {globalFunctions} from '../runtime/globalFunctions'
