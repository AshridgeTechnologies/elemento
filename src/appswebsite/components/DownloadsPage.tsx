import React from 'react'
import {Box, Container, Link, Stack, Typography} from '@mui/material'
import AppBar from '../../shared/AppBar'
import {theme} from '../../shared/styling'
import {ThemeProvider} from '@mui/material/styles'

export default function LearnPage() {

    return <ThemeProvider theme={theme}>
        <Box display='flex' flexDirection='column' height='100%' width='100%'>
            <Box flex='0'>
                <AppBar title='Elemento Online - Downloads' userMenu={false}/>
            </Box>
            <Box flex='1' minHeight={0}>
                <Container>
                    <Stack maxWidth={600} spacing={3} mt={3}>
                        <Typography variant='h4'>Elemento Development Server</Typography>
                        <Typography>Please download the correct version for your operating system.</Typography>
                        <Typography>Your computer will complain that the app is not known, approved, signed etc.<br/>
                            We do not currently have the resources to jump through all the hoops required to get the app
                            approved on each operating system,
                            so you will just have to trust us here, and continue past all the warnings.
                        </Typography>
                        <Link href='/devServer/win/ElementoDevServer.exe'><Typography variant='h5'>Windows</Typography></Link>
                        <Link href='/devServer/macos/ElementoDevServer.zip'><Typography variant='h5'>Mac</Typography></Link>
                        <Link href='/devServer/linux/ElementoDevServer'><Typography variant='h5'>Linux</Typography></Link>
                    </Stack>
                </Container>
            </Box>
        </Box>
    </ThemeProvider>
}
