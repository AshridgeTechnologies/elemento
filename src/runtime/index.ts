import * as components from './components'
import {globalFunctions} from './globalFunctions'
export {components, globalFunctions}

export {default as React} from 'react'
export {default as ReactDOM} from 'react-dom/client'
export {default as appFunctions} from './appFunctions'
export {useGetObjectState, useObjectState} from './appData'
export {asArray, codeGenerationError, parentPath} from './runtimeFunctions'
export {default as AppRunner} from './AppRunner'
export {run} from './run'
