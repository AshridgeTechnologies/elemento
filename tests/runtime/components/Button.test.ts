/**
 * @jest-environment jsdom
 */

import {createElement} from 'react'
import {Button} from '../../../src/runtime/components/index'
import {componentJSON, snapshot, valueObj, wait} from '../../testutil/testHelpers'
import userEvent from '@testing-library/user-event'
import {globalFunctions} from '../../../src/runtime/globalFunctions'
import {testContainer} from '../../testutil/rtlHelpers'

const {Log} = globalFunctions

const doIt = () => {}

test('Button element produces output with properties supplied',
    snapshot(createElement(Button, {path: 'app.page1.save', content: 'Click me!', appearance: 'outline', action: () => {doIt()}}))
)

test('Button element produces output with link style',
    snapshot(createElement(Button, {path: 'app.page1.save', content: 'Click me!', appearance: 'link', action: () => {doIt()}}))
)

test('Button element produces output with display true',
    snapshot(createElement(Button, {path: 'app.page1.save', content: 'Click me!', appearance: 'filled', action: () => {doIt()}, display: true}))
)
test('Button element produces output with display false',
    snapshot(createElement(Button, {path: 'app.page1.save', content: 'Click me!', action: () => {doIt()}, display: false}))
)

test('Button element produces output with properties supplied as state values', async () => {
        const element = createElement(Button, {path: 'app.page1.save', content: valueObj('Click me!'), action: () => {doIt()}, display: valueObj(false)})
        await wait(10)
        expect(componentJSON(element)).toMatchSnapshot()
    }
)

test('Button element produces output with default values where properties omitted', async () => {
        const element = createElement(Button, {path: 'app.page1.save', content: 'Click me!'})
        await wait(10)
        expect(componentJSON(element)).toMatchSnapshot()
    }
)

test('Button does action when clicked', async () => {
    let {el, user} = testContainer(createElement(Button, {
        path: 'app.page1.save',
        content: 'Save me!',
        action: () => Log("I'm saved!")
    },))
    const buttonEl = el`button[id="app.page1.save"]`
    const log = jest.spyOn(console, "log").mockImplementation(() => {})
    try {
        await user.click(buttonEl)
        expect(log).toBeCalledWith("I'm saved!")

    } finally {
        log.mockReset();
    }
} )

