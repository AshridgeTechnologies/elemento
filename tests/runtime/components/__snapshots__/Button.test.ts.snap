// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button element produces output with default values where properties omitted 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-disableElevation css-1kf6eet-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  id="app.page1.save"
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Click me!
</button>
`;

exports[`Button element produces output with enabled false 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-disableElevation css-1kf6eet-MuiButtonBase-root-MuiButton-root"
  disabled={true}
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  Click me!
</button>
`;

exports[`Button element produces output with enabled true 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation css-zt059t-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Click me!
</button>
`;

exports[`Button element produces output with link style 1`] = `
<a
  className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-167u9bt-MuiTypography-root-MuiLink-root"
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Click me!
  </p>
</a>
`;

exports[`Button element produces output with link style and styles 1`] = `
<a
  className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-167u9bt-MuiTypography-root-MuiLink-root"
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-dpqqps-MuiTypography-root"
  >
    Click me!
  </p>
</a>
`;

exports[`Button element produces output with properties supplied 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-disableElevation css-1kf6eet-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Click me!
</button>
`;

exports[`Button element produces output with properties supplied as state values 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-disableElevation css-bx9fca-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Click me!
</button>
`;

exports[`Button element produces output with show false 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-disableElevation css-bx9fca-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Click me!
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`Button element produces output with show true 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation css-zt059t-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Click me!
</button>
`;

exports[`Button element produces output with styles supplied 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-disableElevation css-1bh4wyw-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  id="app.page1.save"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Click me!
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
